### PR TITLE
Added ML Services

### PR DESCRIPTION
**Pull request by jasonsattler**
I am going to test this one more time as part of upgrading from v0.1.4.  

The idea is that they have to create the loadbalancer with the module `ml_network`.
Then all the output from that will get plugged into each service that they want to create.

The terraform code for all the services are the same so that is why I added a `ml_cluster` that will create the ASG, SG, IAM, etc..

Each service the only real difference is the cloud-init script that we run might be different or might need a RDS (Faces) instance.  So that is still possible to bolt on.  Hopefully this makes adding new ML services much easier for us.

### CODE REVIEW

File: README-MLservices.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
+# Graymeta Machine Learning services
+This is optional but is required if you want to use any Graymeta Machine Learning services.
+
+**NOTE:** *The first module below is required to setup a internal load balancer to the different ML services.*
```
Comment (jasonhancock) [inline README-MLservices.md]: s/setup a internal/setup an internal/

File: README-MLservices.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+**UPGRADE:** *If upgrading Faces from v0.1.4 or lower make sure you copy the `RDS Configuration` to the new module and that the module does not change.  Also credits and slates was included with the old faces module.  Now they are seperate module and make sure you update the endpoint/hostname in the extractor config.*
```
Comment (jasonhancock) [inline README-MLservices.md]: s/credits and slates was/credits and slates were/
s/now they are separate module/now they are separate modules/

what does it mean that the "module does not change"?

File: README-MLservices.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+```
+
+## To setup the endpoint for the extractors
+#### GrayMeta Facial Recognition extractor
+For the Graymeta Facial Recognigtion extractor you will need to add the following variables to the platform module.
+```
+module "platform" {
+  source = "github.com/graymeta/terraform-aws-platform?ef=v0.1.5"
+  ...
+  # (Optional) Graymeta ML Services
```
Comment (jasonhancock) [inline README-MLservices.md]: should this soccer reference be removed since it's discussed below in the `All the other extractors` section?

File: README-MLservices.md (hunk 2 above)
Comment (jasonsattler) [reply README-MLservices.md]: Let me rephrase s/module does not change/module name does not change/.
As long as the module name does not change the RDS will not have to be destroyed and recreated.

File: README-MLservices.md (hunk 3 above)
Comment (jasonsattler) [reply README-MLservices.md]: soccer?  Are you asking about the 
```
  # (Optional) Graymeta ML Services
  mlservices_endpoint = "${module.ml_network.endpoint}"
```
So that ml_services_endpoint is what goes into the no_proxy on services and ecs.  Let my try something different here.

**Comment by jasonsattler** [comment]
Well most of the user data is the same except for nld and faces.  Let me try something to make all the common ones a template.

**Comment by jasonsattler** [comment]
I did not want to put the userdata in ml_cluster module since I really wanted that to be generic to create the cluster and hook that all up to the load balancer.  So I created another module to do the common userdata.  That will work for all the services except for Faces and NLD.
